src/traces/scattermapbox/plot.js

```diff
@@ -42,11 +50,21 @@ function ScatterMapbox(subplot, uid) {
 
 var proto = ScatterMapbox.prototype;
 
-proto.addSource = function(k, opts) {
-    this.subplot.map.addSource(this.sourceIds[k], {
-        type: 'geojson',
-        data: opts.geojson
-    });
+proto.addSource = function(k, opts, cluster) {
+    if(cluster.enabled) {
+        this.subplot.map.addSource(this.sourceIds[k], {
+            type: 'geojson',
+            data: opts.geojson,
+            cluster: true,
+            clusterMaxZoom: cluster.maxZoom,
+            clusterRadius: cluster.radius,
+        });
+    } else {
+        this.subplot.map.addSource(this.sourceIds[k], {
+            type: 'geojson',
+            data: opts.geojson,
+        });
+    }
```

Comment:
The changes here are looking good. Thanks for merging two prototypes.
But we could still improve it a bit.
```js
proto.addSource = function(k, opts, cluster) {
    var sourceOpts =
            type: 'geojson',
            data: opts.geojson,
    };
    
    if(cluster.enabled) {
        Lib.extendFlat({}, sourceOpts, {
            cluster: true, // better to rename this key to clusterEnabled?
            clusterMaxZoom: cluster.maxZoom,
            clusterRadius: cluster.radius,
        });
    } 
    
    this.subplot.map.addSource(this.sourceIds[k], sourceOpts);
};
```